server/authz/policy_test.go

```diff
@@ -500,6 +500,101 @@ func TestAuthorizeSoftwareInventory(t *testing.T) {
+	runTestCases(t, []authTestCase{
+		{user: nil, object: noTeamInstaller, action: read, allow: false},
+		{user: nil, object: noTeamInstaller, action: write, allow: false},
+		{user: nil, object: team1Installer, action: read, allow: false},
+		{user: nil, object: team1Installer, action: write, allow: false},
+		{user: nil, object: team2Installer, action: read, allow: false},
+		{user: nil, object: team2Installer, action: write, allow: false},
+
+		{user: test.UserNoRoles, object: noTeamInstaller, action: read, allow: false},
+		{user: test.UserNoRoles, object: noTeamInstaller, action: write, allow: false},
+		{user: test.UserNoRoles, object: team1Installer, action: read, allow: false},
+		{user: test.UserNoRoles, object: team1Installer, action: write, allow: false},
+		{user: test.UserNoRoles, object: team2Installer, action: read, allow: false},
+		{user: test.UserNoRoles, object: team2Installer, action: write, allow: false},
+
+		{user: test.UserAdmin, object: noTeamInstaller, action: read, allow: true},
+		{user: test.UserAdmin, object: noTeamInstaller, action: write, allow: true},
+		{user: test.UserAdmin, object: team1Installer, action: read, allow: true},
+		{user: test.UserAdmin, object: team1Installer, action: write, allow: true},
+		{user: test.UserAdmin, object: team2Installer, action: read, allow: true},
+		{user: test.UserAdmin, object: team2Installer, action: write, allow: true},
+
+		{user: test.UserMaintainer, object: noTeamInstaller, action: read, allow: true},
+		{user: test.UserMaintainer, object: noTeamInstaller, action: write, allow: true},
+		{user: test.UserMaintainer, object: team1Installer, action: read, allow: true},
+		{user: test.UserMaintainer, object: team1Installer, action: write, allow: true},
+		{user: test.UserMaintainer, object: team2Installer, action: read, allow: true},
+		{user: test.UserMaintainer, object: team2Installer, action: write, allow: true},
+
+		{user: test.UserObserver, object: noTeamInstaller, action: read, allow: true},
+		{user: test.UserObserver, object: noTeamInstaller, action: write, allow: false},
+		{user: test.UserObserver, object: team1Installer, action: read, allow: true},
+		{user: test.UserObserver, object: team1Installer, action: write, allow: false},
+		{user: test.UserObserver, object: team2Installer, action: read, allow: true},
+		{user: test.UserObserver, object: team2Installer, action: write, allow: false},
+
+		{user: test.UserObserverPlus, object: noTeamInstaller, action: read, allow: true},
+		{user: test.UserObserverPlus, object: noTeamInstaller, action: write, allow: false},
+		{user: test.UserObserverPlus, object: team1Installer, action: read, allow: true},
+		{user: test.UserObserverPlus, object: team1Installer, action: write, allow: false},
+		{user: test.UserObserverPlus, object: team2Installer, action: read, allow: true},
+		{user: test.UserObserverPlus, object: team2Installer, action: write, allow: false},
+
+		// TODO: confirm gitops permissions
```

Comment:
We'll need to confirm that gitops flows work as expected without read permissions or implement special authz for that piece.